src/interfaces/NamingService.ts

```diff
@@ -0,0 +1,125 @@
+import { Provider } from '..';
+import {  NamingServiceName } from '../publicTypes';
+
+export default interface NamingService {
+  readonly name: NamingServiceName;
+  readonly network: number;
+  readonly url: string | undefined;
+  readonly registryAddress?: string;
+  readonly provider: Provider;
+
+  owner(domain: string): Promise<string>;
+  resolver(domain: string): Promise<string>;
+  namehash(domain: string): string;
+  isSupportedDomain(domain: string): boolean;
+  record(domain: string, key: string): Promise<string>;
+  records(domain: string, keys: string[]): Promise<Record<string, string>>;
+  serviceName(domain: string):NamingServiceName;
```

Comment:
@JohnnyJumper Why do we need domain name to get service name? It's a static property as far as i know.